diff --git a/absl/log/internal/log_message.cc b/absl/log/internal/log_message.cc
--- a/absl/log/internal/log_message.cc
+++ b/absl/log/internal/log_message.cc
@@ -364,7 +364,7 @@ LogMessage& LogMessage::ToSinkOnly(absl::LogSink* absl_nonnull sink) {
   return *this;
 }
 
-#ifdef __ELF__
+#if defined(__ELF__) && !defined(__SYMBIAN32__)
 extern "C" void __gcov_dump() ABSL_ATTRIBUTE_WEAK;
 extern "C" void __gcov_flush() ABSL_ATTRIBUTE_WEAK;
 #endif
@@ -377,7 +377,7 @@ void LogMessage::FailWithoutStackTrace() {
   __debugbreak();
 #endif
 
-#ifdef __ELF__
+#if defined(__ELF__) && !defined(__SYMBIAN32__)
   // For b/8737634, flush coverage if we are in coverage mode.
   if (&__gcov_dump != nullptr) {
     __gcov_dump();
